pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -196,7 +200,7 @@ func (s *Scheduler) add(pod *v1.Pod) error {
 			}
 		}
 
-		node := NewNode(nodeTemplate, s.topology, s.daemonOverhead[nodeTemplate], instanceTypes)
+		node := NewNode(s.ctx, s.kubeClient, nodeTemplate, s.topology, s.daemonOverhead[nodeTemplate], instanceTypes)
```

Comment:
Why are we using a different context object here? How does this differ from `ctx` on line 103?

Reply:
It just wasn't passed down and was only used for logging. Modified to pass it through add().